sympy/tensor/indexed.py

```diff
@@ -663,7 +663,7 @@ def __new__(cls, label, range=None, **kw_args):
                     raise TypeError("Idx object requires integer bounds.")
             args = label, Tuple(*range)
         elif isinstance(range, Expr):
-            if not (range.is_integer or range is S.Infinity):
+            if fuzzy_and([fuzzy_not(range.is_integer), fuzzy_not(range is S.Infinity)]):
```

Comment:
Since `range is S.Infinity` is a normal bool rather than a fuzzy bool it isn't necessary to use `fuzzy_not` for that so this could be
```
if range is not S.Infinity and fuzzy_not(range.is_integer):
```